mindsdb_server/namespaces/datasource.py

```diff
@@ -208,97 +308,147 @@ class Analyze(Resource):
     def get(self, name):
         global global_mdb
         ds = get_datasource(name)
-        if ds['name'] is None:
+        if ds is None:
             print('No valid datasource given')
             abort(400, 'No valid datasource given')
 
         if 'analysis_data' in ds and ds['analysis_data'] is not None:
             return ds['analysis_data'], 200
 
-        analysis = global_mdb.analyse_dataset(ds['source'], sample_margin_of_error=0.025)
+        analysis = get_analysis(ds['source'])
 
         ds['analysis_data'] = analysis
         save_datasource_metadata(ds)
         
         return analysis, 200
 
 
+@ns_conf.route('/<name>/analyze_subset')
+@ns_conf.param('name', 'Datasource name')
+class AnalyzeSubset(Resource):
```

Comment:
A lot of the logic here seems to be similar to that in `DatasourceData` maybe it's worth trying to see if you can reduce the boilerplate a bit more? But maybe there's something making that hard to do which I'm not seeing, so up to you.

Reply:
that not very same, but i agree, it will be better to readability. Done